test/client.test.js

```diff
@@ -5,7 +5,13 @@ const proxyquire = require("proxyquire").noCallThru();
 const sinon = require("sinon");
 const realRequest = require("../dist/request");
 
+const fail = result =>
+  assert.fail(`expected promise to be rejected, got resolved with ${result}`);
```

Comment:
nitpick: `${result}` will result in `[object Object]` when `result` is an object. How about `${util.inspect(result)}`?